### PR TITLE
enable initial build for rocm for fp6_llm

### PR DESCRIPTION
**Pull request by lcskrishna**
Initial enablement task for FP6_llm.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -71,15 +73,14 @@ def get_extensions():
                 "-O3" if not debug_mode else "-O0",
                 "-fdiagnostics-color=always",
             ],
-            "nvcc": [
-                "-O3" if not debug_mode else "-O0",
-                "-t=0",
-            ]
         }
+        if use_cuda and not IS_ROCM:
```
Comment (petrex) [inline setup.py:77]: What would be the compile flags for rocm?

File: setup.py [hunk 2]
```diff
@@ -107,17 +108,35 @@ def get_extensions():
     extensions_cuda_dir = os.path.join(extensions_dir, "cuda")
     cuda_sources = list(glob.glob(os.path.join(extensions_cuda_dir, "**/*.cu"), recursive=True))
 
-    if use_cuda:
+    extensions_hip_dir = os.path.join(extensions_dir, "cuda", "fp6_llm")
+    hip_sources = list(glob.glob(os.path.join(extensions_hip_dir, "*.cu"), recursive=True))
```
Comment (petrex) [inline setup.py:112]: seems file for now, but we might have other extensions other than ".cu" later?

File: setup.py [hunk 3]
```diff
@@ -107,17 +108,35 @@ def get_extensions():
...
-            sources,
-            extra_compile_args=extra_compile_args,
-            extra_link_args=extra_link_args,
-        )
-    ]
+    # TOOD: Remove this and use what CUDA has once we fix all the builds.
+    if IS_ROCM and use_cuda:
+        sources += hip_sources
+
+    ## TODO: remove this condition and use what we have in CUDA once we fix the individual builds.
```
Comment (petrex) [inline setup.py:121]: understand we are wip, might be able to consolidate the code later?

**Comment by petrex** [comment]
One other question: what would be the entry point for hipify_torch?

**Comment by petrex** [comment]
also I've created an staging branch for this PR : `rocm_enablement_staging` , let's keep `main ` sync with upstream?

File: setup.py (hunk 2 above)
Comment (lcskrishna) [reply setup.py:112]: yes, once we enable all the extensions - we can remove hip_sources and start using cuda_sources directly.

File: setup.py (hunk 3 above)
Comment (lcskrishna) [reply setup.py:121]: yes

**Comment by lcskrishna** [comment]
> also I've created an staging branch for this PR : `rocm_enablement_staging` , let's keep `main ` sync with upstream?

Updated the PR to use rocm_enablement_staging. That's a good idea to use main for syncing with upstream.

**Comment by lcskrishna** [comment]
> One other question: what would be the entry point for hipify_torch?

Entry points are CUDAExtension. Torch code has integrated hipify_torch as part of CUDAExtensions.

**Pull request merged**